src/Concordium/Client/Runner.hs

```diff
@@ -2165,14 +2165,9 @@ getWasmModule :: (MonadIO m)
               -> Text -- ^Hash of the block to query in.
               -> ClientMonad m Wasm.WasmModule
 getWasmModule namedModRef block = do
-  res <- getModuleSource (Text.pack . show $ nmrRef namedModRef) block
-
-  case res of
-    Left err -> logFatal ["I/O error:", err]
-    Right (GRPCResponse _ "") -> logFatal [[i|the module reference #{showNamedModuleRef namedModRef} does not exist in block #{block}|]]
-    Right unparsedWasmMod -> case S.decode $ grpcResponseVal unparsedWasmMod of
-      Left err' -> logFatal [[i|could not decode Wasm Module:|], err']
-      Right wasmMod -> return wasmMod
+  bh <- readOrFail block
+  getModuleSourceV2 (nmrRef namedModRef) (Given bh) >>=
```

Comment:
Again, what will be printed to the user when the module doesn't exist?

Reply:
The node response stating the nature of the error.

Reply:
Again, this error is just worse than the one we had. So please make the error at least as good as it was.